packages/base/src/util/AriaLabelHelper.ts

```diff
@@ -57,7 +103,105 @@ const getAssociatedLabelForTexts = (el: HTMLElement) => {
+		if (!registry) {
+			return;
+		}
+		const oldAssociatedElements = registry.trackedElements;
+		const newAssociatedElements = _getAllAssociatedElementsFromDOM(el, trackedProperties);
+		oldAssociatedElements.forEach(oldElement => {
+			if (!newAssociatedElements.includes(oldElement)) {
+				_removeRegistryTrackedElement(registry, oldElement);
+			}
+		});
+		newAssociatedElements.forEach(newElement => {
+			if (!oldAssociatedElements.includes(newElement)) {
+				_addRegistryTrackedElement(registry, newElement);
+				registry.trackedElements.push(newElement);
+			}
+		});
+		registry?.callback();
+	};
+	return invalidationCallback;
+};
+
+const registerUI5Element = (el: UI5Element, callback: () => void, trackedProperties: string[]) => {
+	if (registredElements.has(el)) {
+		return;
+	}
+	const allAssociatedElements = _getAllAssociatedElementsFromDOM(el, trackedProperties);
+	const invalidationCallback = _createInvalidationCallback(el, trackedProperties);
+	const registry = {
+		elementId: el.id,
+		trackedElements: allAssociatedElements,
+		trackedProperties,
+		callback,
+		invalidationCallback,
+	};
+	registredElements.set(el, registry);
+	el.attachInvalidate(invalidationCallback);
+
+	allAssociatedElements.forEach((element: HTMLElement) => {
+		_addRegistryTrackedElement(registry, element);
+	});
+	callback();
+};
+
+const _addRegistryTrackedElement = (registry:RegistedElement, trackedElement: HTMLElement) => {
+	let associatedElement = associatedElements.get(trackedElement);
+	if (!associatedElement) {
+		associatedElement = { observer: null, callbacks: [] };
+		const observer = new MutationObserver(() => {
+			const callbacks = (associatedElement as AssociatedElement).callbacks;
+			callbacks.forEach(callback => {
+				callback();
+			});
+			const domEl = document.getElementById(trackedElement.id);
+			// if no longer should be tracked from this registry, remove it
+			if (!domEl || registry.elementId !== trackedElement.getAttribute("for")) {
+				_removeRegistryTrackedElement(registry, trackedElement);
+			}
+		});
+		associatedElement.observer = observer;
+		observer.observe(trackedElement, observerOptions);
+		associatedElements.set(trackedElement, associatedElement);
+	}
+	if (!associatedElement.callbacks.includes(registry.callback)) {
+		associatedElement.callbacks.push(registry.callback);
+	}
+};
+
+const _removeRegistryTrackedElement = (registry:RegistedElement, trackedElement: HTMLElement) => {
+	const associatedElement = associatedElements.get(trackedElement);
+	if (associatedElement) {
+		associatedElement.callbacks = associatedElement.callbacks.filter(itm => itm !== registry.callback);
+		if (!associatedElement.callbacks.length) {
+			associatedElement.observer?.disconnect();
+			associatedElements.delete(trackedElement);
+		}
+	}
+	registry.trackedElements = registry.trackedElements.filter(itm => itm !== trackedElement);
+};
+
+const unregisterUI5Element = (el: UI5Element) => {
+	const registry = registredElements.get(el);
```

Comment:
`const registry` => `const registeredElement` (wrong semantics, as `registeredElements` is the registry itself, so one of the elements from it should not be named registry too)